solr/core/src/java/org/apache/solr/cloud/overseer/CollectionMutator.java

```diff
@@ -136,8 +154,13 @@ public ZkWriteCommand modifyCollection(final ClusterState clusterState, ZkNodePr
       return ZkStateWriter.NO_OP;
     }
 
-    return new ZkWriteCommand(coll.getName(),
-        new DocCollection(coll.getName(), coll.getSlicesMap(), m, coll.getRouter(), coll.getZNodeVersion(), coll.getZNode()));
+    DocCollection collection = new DocCollection(coll.getName(), coll.getSlicesMap(), m, coll.getRouter(), coll.getZNodeVersion(), coll.getZNode());
+    if (replicaOps == null){
+      return new ZkWriteCommand(coll.getName(), collection);
+    } else {
+      return new ZkWriteCommand(coll.getName(), collection, replicaOps, true);
+    }
```

Comment:
The two return statements above end up doing exactly the same thing.

Reply:
My bad, the two are different.
When `replicaOps` is `null`, `ops` in `ZkWriteCommand` is set to `PerReplicaStates.WriteOps.touchChildren()`.
When `replicaOps` is not `null`, `ops` in `ZkWriteCommand` is set to `replicaOps`.